File: src/my_class_pkg/scripts/ros_client.py
#!/usr/bin/env python3
import rospy
from my_class_pkg.srv import MyServiceMsg

if __name__ == "__main__":
    # Initialize ROS node
    rospy.init_node("python_service_client")
    # Wait for service to start (avoid call before server is ready)
    rospy.wait_for_service("my_service")
    try:
        # Create service client
        client = rospy.ServiceProxy("my_service", MyServiceMsg)
        # Send request (input=2)
        response = client(2)
        rospy.loginfo(f"Response received: output = {response.output}")
    except rospy.ServiceException as e:
        rospy.logerr(f"Service call failed: {e}")

File: src/my_class_pkg/src/ros_client.cpp
#include "ros/ros.h"
#include "my_class_pkg/MyServiceMsg.h"

int main(int argc, char **argv) {
    ros::init(argc, argv, "service_client");
    ros::NodeHandle nh;
    ros::ServiceClient client = nh.serviceClient<my_class_pkg::MyServiceMsg>("my_service");
    
    my_class_pkg::MyServiceMsg srv;
    srv.request.input = 42;
    
    if (client.call(srv)) {
        ROS_INFO("Response received: output = %ld", (long int)srv.response.output);
    } else {
        ROS_ERROR("Service call failed!");
        return 1;
    }
    return 0;
}
